enricher/api/src/main/java/io/fabric8/maven/enricher/api/BaseEnricher.java

```diff
@@ -114,4 +127,32 @@ protected boolean isOpenShiftMode() {
         }
         return false;
     }
+
+    protected void ensureMetadata(PodTemplateSpecBuilder obj) {
+        ObjectMeta metadata = obj.getMetadata();
+        if (metadata == null) {
+            obj.withNewMetadata().endMetadata();
+        }
+    }
+
+    protected void addInitContainer(ObjectMetaBuilder metadata, JSONObject initContainer) {
+        String initContainerAnnotation = metadata.getAnnotations().get(INIT_CONTAINER_ANNOTATION);
```

Comment:
annotations are always non-null, right ? 

I always get confused with when using builder and when using object directly. I.e. I would have expected here a 'editAnnotations()' or so. But I remember darkly that for lists one always use the 'direct' objects.